lib/src/components/sprite_component.dart

```diff
@@ -24,14 +24,19 @@ class SpriteComponent extends PositionComponent {
   Paint overridePaint;
 
   /// Creates a component with an empty sprite which can be set later
-  SpriteComponent();
-
-  SpriteComponent.fromImage(Vector2 size, Image image)
-      : this.fromSprite(size, Sprite(image));
-
-  SpriteComponent.fromSprite(Vector2 size, this.sprite) {
-    super.size.setFrom(size);
-  }
+  SpriteComponent({
+    Vector2 position,
+    Vector2 size,
+    this.sprite,
+    this.overridePaint,
+  }) : super(position: position, size: size);
+
+  factory SpriteComponent.fromImage(
+    Vector2 size,
+    Image image, {
+    Vector2 position,
+  }) =>
+      SpriteComponent(size: size, sprite: Sprite(image), position: position);
```

Comment:
nit: I prefer block bodies if the thing won't fit on the same line as the function dec